ci/integTestGen/src/integTestGen.jl

```diff
@@ -0,0 +1,265 @@
+            for dependency_name_version in keys(project_tree[project_name_version])
+                # dependency matches, add to packages
+                if startswith(dependency_name_version, package_name)
+                    push!(packages, project_name)
+                    break
+                end
+            end
+            # independent of a match, under investigate all dependencies too, because they can also have the package as dependency
+            traverse_tree!(package_name, package_prefix, project_tree[project_name_version], packages, visited_packages)
+        end
+    end
+end
+
+"""
+    generate_job_yaml!(package_name::String, job_yaml::Dict)
+
+Generate GitLab CI job yaml for integration test of a give package.
+
+# Args
+- `package_name::String`: Name of the package to test.
+- `job_yaml::Dict`: Add generated job to this dict.
+"""
+function generate_job_yaml!(package_name::String, job_yaml::Dict)
+    package_info = package_infos[package_name]
+    # if modified_url is empty, use original url
+    if package_info.modified_url == ""
+        url = package_info.url
+    else
+        url = package_info.modified_url
+    end
+
+    script = [
+        "apt update",
+        "apt install -y git",
+        "cd /"
+    ]
+
+
+    split_url = split(url, "#")
+    if length(split_url) > 2
+        error("Ill formed url: $(url)")
+    end
+
+    push!(script, "git clone $(split_url[1]) integration_test")
+    push!(script, "cd integration_test")
+
+    if length(split_url) == 2
+        push!(script, "git checkout $(split_url[2])")
+    end
+
+    push!(script, "julia --project=. -e 'import Pkg; Pkg.Registry.add(Pkg.RegistrySpec(url=\"https://github.com/QEDjl-project/registry.git\"));'")
+    push!(script, "julia --project=. -e 'import Pkg; Pkg.Registry.add(Pkg.RegistrySpec(url=\"https://github.com/JuliaRegistries/General\"));'")
+    ci_project_dir = ENV["CI_PROJECT_DIR"]
+    push!(script, "julia --project=. -e 'import Pkg; Pkg.develop(path=\"$ci_project_dir\");'")
+    push!(script, "julia --project=. -e 'import Pkg; Pkg.test(; coverage = true)'")
+
+    job_yaml["IntegrationTest$package_name"] = Dict(
+        "image" => "julia:1.9",
+        "interruptible" => true,
+        "tags" => ["cpuonly"],
+        "script" => script)
+end
+
+"""
+    generate_dummy_job_yaml!(job_yaml::Dict)
+
+Generates a GitLab CI dummy job, if required.
+
+# Args
+- `job_yaml::Dict`: Add generated job to this dict.
+"""
+function generate_dummy_job_yaml!(job_yaml::Dict)
+    job_yaml["DummyJob"] = Dict("image" => "alpine:latest",
+        "interruptible" => true,
+        "script" => ["echo \"This is a dummy job so that the CI does not fail.\""])
+end
+
+if abspath(PROGRAM_FILE) == @__FILE__
+    # custom commit message variable can be set as first argument
+    if length(ARGS) < 1
+        extract_env_vars_from_git_message!()
+    else
+        extract_env_vars_from_git_message!(ARGS[1])
+    end
+
+    set_modified_package_url!()
```

Comment:
Here, it is not clear, which variable will be changed! 
It is more safe if the variable to be mutated is passed as an argument to the function. See the comment on the function itself for more details.